### PR TITLE
Plugin migration out of the mono-repo

### PR DESCRIPTION
**Pull request by kdeldycke**
This PR:
* [x] Move the `asset` plugin from the monorepo
* [x] Repackage the whole thing following latest [`cookiecutter` template](https://github.com/getpelican/cookiecutter-pelican-plugin)

### CODE REVIEW

File: pelican/plugins/webassets/assets.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            generator.env.assets_environment.config[item[0]] = item[1]
+
+    if 'ASSET_BUNDLES' in generator.settings:
+        for name, args, kwargs in generator.settings['ASSET_BUNDLES']:
+            generator.env.assets_environment.register(name, *args, **kwargs)
+
+    if 'ASSET_DEBUG' in generator.settings:
+        generator.env.assets_environment.debug = generator.settings['ASSET_DEBUG']
+    elif logging.getLevelName(logger.getEffectiveLevel()) == "DEBUG":
+        generator.env.assets_environment.debug = True
```
Comment (bryanbrattlof) [inline pelican/plugins/webassets/assets.py]: How do we want to handle the config variable name changes between the versions? Should we use both and deprecate one for a few years before removing?

Comment (kdeldycke) [reply pelican/plugins/webassets/assets.py]: I guess we can support both for a while... Haven't looked in the detailed code for the moment by the way. 

But maybe we can split the work:
* let me repackage the legacy plugin in this PR
* merge it back to `main`
* then let you merge your code here

What do you think?

Comment (bryanbrattlof) [reply pelican/plugins/webassets/assets.py]: Sounds like a good plan to me :+1:

Comment (kdeldycke) [reply pelican/plugins/webassets/assets.py]: Let's do it! :)

**Comment by kdeldycke** [comment]
This PR is ready to be reviewed/merged to complete the initial migration out of the mono-repo! :)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Contributions are welcome and much appreciated. Every little bit helps. You can contribute by improving the documentation, adding missing features, and fixing bugs. You can also help out by reviewing and commenting on [existing issues][].
+
+To start contributing to this plugin, review the [Contributing to Pelican][] documentation, beginning with the **Contributing Code** section.
+
+[existing issues]: https://github.com/pelican-plugins/webassets/issues
+[Contributing to Pelican]: https://docs.getpelican.com/en/latest/contribute.html
+
+License
+-------
```
Comment (kernc) [inline README.md]: An off title. Otherwise :clap: :1st_place_medal:

**Review by kernc — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kdeldycke) [reply README.md]: Nice catch! 😁

Thanks for the review @kernc ! 😊

**Review by avaris — APPROVED** [review]

**Comment by kdeldycke** [comment]
With 3 validations, I guess we're good to go. Let's merge this ! :)

**Comment by kdeldycke** [comment]
@bryanbrattlof, you're free to bring in your improvements by the way of a PR! :)

**Pull request merged**